File: app/src/main/java/by/offvanhooijdonk/plaincalendar/widget/view/configure/IConfigureView.java
package by.offvanhooijdonk.plaincalendar.widget.view.configure;

import java.util.List;

import by.offvanhooijdonk.plaincalendar.widget.model.CalendarModel;
import by.offvanhooijdonk.plaincalendar.widget.model.WidgetModel;

public interface IConfigureView {
    void displayCalendarsDialog(List<CalendarModel> calendars);

    void onWidgetSettingsLoaded(WidgetModel widget);

    void showCalendarsLoadProgress(boolean isShow);

    void showNoWidgets(boolean isShow);

    //void onCalendarSettingsSaved();

    void notifyChangesAndFinish();
}

File: app/src/main/java/by/offvanhooijdonk/plaincalendar/widget/view/configure/IConfigureView.kt
package by.offvanhooijdonk.plaincalendar.widget.view.configure

import by.offvanhooijdonk.plaincalendar.widget.model.CalendarModel
import by.offvanhooijdonk.plaincalendar.widget.model.WidgetModel

interface IConfigureView {
    fun displayCalendarsDialog(calendars: List<CalendarModel>)
    fun onWidgetSettingsLoaded(widget: WidgetModel)
    fun showCalendarsLoadProgress(isShow: Boolean)
    fun showNoWidgets(isShow: Boolean)

    fun notifyChangesAndFinish()
}
